PlutoHelperAgent/AppDelegate.m

```diff
@@ -193,7 +221,52 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
         BOOL extensionGood = [WhiteListProcessor checkIsInWhitelist:projectPath whiteListName:@"extensions_list" prefix:false];
               
         if (pathGood && extensionGood) {
-            [self tryOpenProject:projectPath];
+            NSArray *pathParts = [projectPath componentsSeparatedByString:@"?"];
+            NSString *partZero = [pathParts objectAtIndex:0];
+            if ((urlParams[@"premiereVersion"]) && (!urlParams[@"force"])) {
+                NSDictionary * premiereVersions = [[NSUserDefaults standardUserDefaults] objectForKey:@"Premiere_versions"];
+                if (premiereVersions[urlParams[@"premiereVersion"]]) {
+                    NSTask *openTask = [[NSTask alloc] init];
+                    [openTask setLaunchPath:@"/usr/bin/open"];
+                    [openTask setCurrentDirectoryPath:@"/"];
+                    [openTask setArguments:@[ premiereVersions[urlParams[@"premiereVersion"]], partZero ]];
+                    [openTask launch];
+                } else {
+                    NSArray *versionsArray = [premiereVersions allKeys];
+                    NSMutableArray *processedVersionsArray = [NSMutableArray new];
+                    for (id version in versionsArray) {
+                        NSString *versionForProcessing = version;
+                        NSUInteger numberOfOccurrencesInVersion = [[versionForProcessing componentsSeparatedByString:@"."] count] - 1;
+                        if (numberOfOccurrencesInVersion == 0) {
+                            versionForProcessing = [NSString stringWithFormat:@"%@00", versionForProcessing];
+                        } else if (numberOfOccurrencesInVersion == 1) {
+                            versionForProcessing = [NSString stringWithFormat:@"%@0", versionForProcessing];
+                        }
+                        versionForProcessing = [versionForProcessing stringByReplacingOccurrencesOfString:@"." withString:@""];
+                        NSUInteger versionStringLength = [versionForProcessing length];
+                        if (versionStringLength == 3) {
+                            versionForProcessing = [NSString stringWithFormat:@"0%@", versionForProcessing];
+                        }
+                        [processedVersionsArray addObject:versionForProcessing];
+                    }
+                    NSArray *sortedProcessedVersionsArray = [processedVersionsArray sortedArrayUsingComparator:
```

Comment:
I would put the version management stuff - sorting the lists, parsing out, etc. into its own object (.m / .h file) again to make it more readable

Reply:
Remember that you can declare `static` methods (i.e. methods that don't require you to initialise an object first) with `+` sign instead of a `-` like this:

```objective-c
+ someMethod:(NSString *)someArg
```

Reply:
I have put this in its own function.

Reply:
Great, so now you just need to move it to another object and do the tests.

Reply:
Done.